web/src/stream_data.ts

```diff
@@ -619,7 +623,14 @@ export function can_access_topic_history(sub: StreamSubscription): boolean {
 }
 
 export function can_preview(sub: StreamSubscription): boolean {
-    return sub.subscribed || !sub.invite_only || sub.previously_subscribed;
+    if (sub.previously_subscribed) {
```

Comment:
Is this meant to be `sub.subscribed`? I'm actually not sure why `previously_subscribed` should be offered here -- that feels to me likely to be a bug in the previous code.

Reply:
If a user is part of a private channel, they receive 3 messages before they get kicked out. The channel will show up as archived and those 3 messages would be visible to the user. That's what `previously_subscribed` checks for. The user would be able to preview just those messages they have access to.

If we want to use it in `can_preview` or not is another question.

Reply:
That behavior is changing in #29380, which I'm expecting we will include in 10.0. So I think skipping `previously_subscribed` here is correct. And of course, we do need `subscribed` to still be listed here.

Reply:
Okay, fixed [here](https://github.com/zulip/zulip/compare/87e2a66431fa20417b81bb42e4c2895c022a5b3e..071fb9663ae7d2b4a8fc1e0978be5cd8d6e4801c)